content/tasks.go

```diff
@@ -39,6 +41,42 @@ func NewTaskMarker(t TaskStatus) *TaskMarker {
 	}
 }
 
+// ParseTaskStatus parses a task status from a TO DO/DOING/DONE/etc. string.
+func ParseTaskStatus(status string) TaskStatus {
+	caseInsensitiveStatus := strings.ToUpper(status)
+	switch caseInsensitiveStatus {
+	case "TODO":
+		return TaskStatusTodo
+	case "DONE":
+		return TaskStatusDone
+	case "DOING":
+		return TaskStatusDoing
+	case "LATER":
+		return TaskStatusLater
+	case "NOW":
+		return TaskStatusNow
+	case "CANCELLED":
+		return TaskStatusCancelled
+	case "CANCELED":
+		return TaskStatusCanceled
+	case "IN-PROGRESS":
+		return TaskStatusInProgress
+	case "WAIT":
+		return TaskStatusWait
+	case "WAITING":
+		return TaskStatusWaiting
+	default:
+		return TaskStatusNone
+	}
+}
+
+// NewTaskMarkerFromString creates a new task marker from a TO DO/DOING/DONE/etc. string.
+func NewTaskMarkerFromString(t string) *TaskMarker {
```

Comment:
I tried to follow the naming in your package but I don't know if these function names were good choices.
Please suggest better ones if you have them.